ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisherTest.java

```diff
@@ -103,26 +118,40 @@ public void setUp() {
     publisher = new PullQueryPublisher(
         engine,
         serviceContext,
+        exec,
         statement,
         Optional.empty(),
         TIME_NANOS,
         routingFilterFactory,
         create(1),
         haRouting);
 
-
-    when(statement.getStatementText()).thenReturn("");
     when(statement.getSessionConfig()).thenReturn(sessionConfig);
     when(sessionConfig.getConfig(false)).thenReturn(ksqlConfig);
     when(sessionConfig.getOverrides()).thenReturn(ImmutableMap.of());
-    when(pullQueryResult.getQueryId()).thenReturn(queryId);
     when(pullQueryResult.getSchema()).thenReturn(PULL_SCHEMA);
-    when(pullQueryResult.getTableRows()).thenReturn(tableRows);
-    when(pullQueryResult.getSourceNodes()).thenReturn(Optional.empty());
-    when(engine.executePullQuery(any(), any(), any(), any(), any(), any()))
+    when(pullQueryResult.getPullQueryQueue()).thenReturn(pullQueryQueue);
+    doNothing().when(pullQueryResult).onException(onErrorCaptor.capture());
+    doNothing().when(pullQueryResult).onCompletion(completeCaptor.capture());
+    int[] times = new int[1];
+    doAnswer(inv -> {
+      Collection<? super KeyValue<List<?>, GenericRow>> c = inv.getArgument(0);
+      if (times[0] == 0) {
+        c.add(KV1);
+      } else if (times[0] == 1) {
+        c.add(KV2);
+        completeCaptor.getValue().accept(null);
+      }
+      times[0]++;
+      return null;
+    }).when(pullQueryQueue).drainTo(any());
```

Comment:
I trust that if you played around with it and it wasn't clear then it would probably require more work to retrofit it than it's worth. I'm happy leaving it as is